src/renderer/widgets/PasswordInputField.cpp

```diff
@@ -134,19 +134,24 @@ bool CPasswordInputField::draw(const SRenderData& data) {
 
     g_pRenderer->renderRect(inputFieldBox, innerCol, inputFieldBox.h / 2.0);
 
-    const int PASS_SIZE    = std::nearbyint(inputFieldBox.h * dt_size * 0.5f) * 2.f;
-    const int PASS_SPACING = std::floor(PASS_SIZE * dt_space);
+    const int PASS_SIZE      = std::nearbyint(inputFieldBox.h * dt_size * 0.5f) * 2.f;
+    const int PASS_SPACING   = std::floor(PASS_SIZE * dt_space);
+    const int DOT_AREA_WIDTH = inputFieldBox.w - PASS_SPACING * 2;          // avail width for dots
+    const int MAX_DOTS       = DOT_AREA_WIDTH / (PASS_SIZE + PASS_SPACING); // max amount of dots that can fit in the area
 
     if (!hiddenInputState.enabled) {
-        for (size_t i = 0; i < std::floor(dots.currentAmount); ++i) {
-            Vector2D currentPos = inputFieldBox.pos() + Vector2D{PASS_SPACING * 2, inputFieldBox.h / 2.f - PASS_SIZE / 2.f} + Vector2D{(PASS_SIZE + PASS_SPACING) * i, 0};
+        size_t numDotsToDraw = std::min(std::floor(dots.currentAmount),
+                                        static_cast<float>(MAX_DOTS)); // Limit the number of dots to draw
+
+        for (size_t i = 0; i < numDotsToDraw; ++i) {
+            Vector2D currentPos = inputFieldBox.pos() + Vector2D{PASS_SPACING, inputFieldBox.h / 2.f - PASS_SIZE / 2.f} + Vector2D{(PASS_SIZE + PASS_SPACING) * i, 0};
             CBox     box{currentPos, Vector2D{PASS_SIZE, PASS_SIZE}};
             g_pRenderer->renderRect(box, fontCol, PASS_SIZE / 2.0);
         }
 
-        if (dots.currentAmount != std::floor(dots.currentAmount)) {
-            Vector2D currentPos = inputFieldBox.pos() + Vector2D{PASS_SPACING * 2, inputFieldBox.h / 2.f - PASS_SIZE / 2.f} +
-                Vector2D{(PASS_SIZE + PASS_SPACING) * std::floor(dots.currentAmount), 0};
+        if (dots.currentAmount != std::floor(dots.currentAmount) && std::floor(dots.currentAmount) < MAX_DOTS) {
+            Vector2D currentPos =
+                inputFieldBox.pos() + Vector2D{PASS_SPACING, inputFieldBox.h / 2.f - PASS_SIZE / 2.f} + Vector2D{(PASS_SIZE + PASS_SPACING) * std::floor(dots.currentAmount), 0};
```

Comment:
uuuh why this format change? maybe slap a clang-format?

Reply:
yeah as in the other pr, clion unfortunatly doesnt do that automatically i need ot manually run it